### ISSUE TITLE
kernel throws "not a text" on lilu plugins with Lilu 1.2.1

### ISSUE THREAD

**Issue by lukrido** [vit9696/Lilu#35]
Hello

I have two plugins working along my lilu kext:
- shiki
- AppleALC

Since I upgraded today from Lilu 1.0.0, I have the following sequence of messages in the Console (2x).
If I remove one plugin, I only have it once.
For AppleALC (latest version), it is working properly, but still shows the warnings.
For Shiki, I have some issues, but I do not think it is related to that.

With Lilu 1.0.0, I only had AppleALC, and it didn't give any message.

> 09/12/17 19:14:54,000 kernel[0]: kext submap [0x<ptr> - 0x<ptr>], kernel text [0x<ptr> - 0x<ptr>]
09/12/17 19:14:54,000 kernel[0]: "name" not a kext
09/12/17 19:14:54,000 kernel[0]: "FailedCLUT" not a kext
09/12/17 19:14:54,000 kernel[0]: "FailedImage" not a kext

**Comment by vit9696** [comment]
Hi,

TL;DR it is not related to Lilu, and it happens when you inject any kexts with a bootloader. Basically it is a bug in Clover, and you should direct your issue to Clover team (or another bootloader dev team if you do not use Clover).

This message happens within XNU `OSKext::createExcludeListFromBooterData`, which iterates over `IODeviceTree:/chosen/memory-map` properties, and sees non-kext properties before the others. `name`, `FailedCLUT`, and `FailedImage` are not kexts, but parameters, e.g. the latter two describe how to visually display boot failure (colour/image).

Why does it do it: If booter kext injection is used XNU expects to find 2 `com.apple.driver.KextExcludeList` kexts instead of 1. Firstly the one from booter, then the actual one from prelinked.

It is not clear why would it insist on all memory-map parameters preceding KextExcludeList to be kexts, since they obviously may not be (all the three are and should be present even on real macs), so you could even file a radar, but I guess it is likely that KextExcludeList will come first anyway and thus avoid the message.

If you just need a workaround, you may try injecting some dummy/empty `com.apple.driver.KextExcludeList` via Clover. From my point of view, this is something Clover should do by default (since clearly XNU expects it), and therefore I consider it a bug.
